### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'results/DeepGraphGO-Model-0-mf-test.npy'

### ISSUE THREAD

**Issue by johnnytam100** [yourh/DeepGraphGO#15]
Hi, I tried to run

`sh ./scripts/run_mf.sh
`

but got 

```
...
[I 220621 15:43:33 psiblast_utils:26] psiblast -out results/mf-valid-ppi-blast-out.xml -outfmt 5 -query data/mf_valid.fasta -db data/ppi_blastdb -evalue 0.001 -num_threads 40 -num_iterations 1
Traceback (most recent call last):
  File "main.py", line 92, in <module>
    main()
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "main.py", line 60, in main
    *_, valid_ppi, valid_y = get_homo_ppi_idx(valid_pid_list, data_cnf['valid']['fasta_file'],
  File "/home/cltam/20220621_DeepGraphGO_experiment/DeepGraphGO/deepgraphgo/data_utils.py", line 126, in get_homo_ppi_idx
    blast_sim = blast(net_blastdb, pid_list, fasta_file, blast_output_path)
  File "/home/cltam/20220621_DeepGraphGO_experiment/DeepGraphGO/deepgraphgo/psiblast_utils.py", line 44, in blast
    return psiblast(*args, **kwargs, num_iterations=1)
  File "/home/cltam/20220621_DeepGraphGO_experiment/DeepGraphGO/deepgraphgo/psiblast_utils.py", line 27, in psiblast
    cline()
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/Bio/Application/__init__.py", line 569, in __call__
    raise ApplicationError(return_code, str(self), stdout_str, stderr_str)
Bio.Application.ApplicationError: Non-zero return code 127 from 'psiblast -out results/mf-valid-ppi-blast-out.xml -outfmt 5 -query data/mf_valid.fasta -db data/ppi_blastdb -evalue 0.001 -num_threads 40 -num_iterations 1', message '/bin/sh: 1: psiblast: not found'
Traceback (most recent call last):
  File "bagging.py", line 38, in <module>
    main()
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "bagging.py", line 31, in main
    sc_mat += np.load(res_path/F'{model_name}-Model-{i}-{data_name}-{test_name}.npy') / num_models
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/numpy/lib/npyio.py", line 416, in load
    fid = stack.enter_context(open(os_fspath(file), "rb"))
FileNotFoundError: [Errno 2] No such file or directory: 'results/DeepGraphGO-Model-0-mf-test.npy'
Usage: evaluation.py [OPTIONS] PID_GO PID_GO_SC
Try 'evaluation.py --help' for help.

Error: Invalid value for 'PID_GO_SC': Path 'results/DeepGraphGO-Ensemble-mf-test.txt' does not exist.
```

and the results folder was empty.

May I know how to solve this?

Thanks!

**Comment by yourh** [comment]
`Bio.Application.ApplicationError: Non-zero return code 127 from 'psiblast -out results/mf-valid-ppi-blast-out.xml -outfmt 5 -query data/mf_valid.fasta -db data/ppi_blastdb -evalue 0.001 -num_threads 40 -num_iterations 1', message '/bin/sh: 1: psiblast: not found'`
You need to install the BLAST tool.

**Comment by johnnytam100** [comment]
Thanks @yourh ! Then I met another error:

```
(deepgraphgo) cltam@luna:~/20220621_DeepGraphGO_experiment/DeepGraphGO$ sh ./scripts/run_mf.sh
[I 220621 18:25:00 main:37] Model: DeepGraphGO, Path: models/DeepGraphGO-Model-0-mf, Dataset: mf
/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/dgl/data/graph_serialize.py:177: DGLWarning: You are loading a grap                                                                                                                                          h file saved by old version of dgl.              Please consider saving it again with the current format.
  dgl_warning(
Traceback (most recent call last):
  File "main.py", line 92, in <module>
    main()
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "main.py", line 46, in main
    dgl_graph.edata['ppi'] = dgl_graph.edata['ppi'].float().cuda()
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/dgl/view.py", line 198, in __setitem__
    self._graph._set_e_repr(self._etid, self._edges, {key : val})
  File "/home/cltam/anaconda3/envs/deepgraphgo/lib/python3.8/site-packages/dgl/heterograph.py", line 4223, in _set_e_repr
    raise DGLError('Cannot assign edge feature "{}" on device {} to a graph on'
dgl._ffi.base.DGLError: Cannot assign edge feature "ppi" on device cuda:0 to a graph on device cpu. Call DGLGraph.to() to copy the gra                                                                                                                                          ph to the same device.

```

Alternatively, may I know if it is possible to obtain from you the pre-computed results for molecular function?

**Comment by yourh** [comment]
I guess a different version of DGL may cause this error.

**Comment by johnnytam100** [comment]
um... my DGL version is `0.4.3post2`, the version in README.

![image](https://user-images.githubusercontent.com/51283097/176335379-cc7facb5-bc66-4c79-a850-cfd42e5946df.png)

Again, is it possible to obtain from you the pre-computed results?

**Comment by yourh** [comment]
I don't really know the reason unless I know more information. I asked several friends to try to run the codes and they didn't find any problems. Here is my running process. 
![image](https://user-images.githubusercontent.com/28811637/176350176-a96c0ac0-5026-436e-8eff-ca3e370db8db.png)
I believe you could get a totally similar result by yourself with these codes. If you really need the pre-computed results, I can provide them when I'm not so busy.

**Comment by johnnytam100** [comment]
I have tried on several systems but failed.
I would appreciate a lot if you can share the precomputed results when you have time.

**Comment by Amels404** [comment]
Hello,  thank you for sharing this. I have the same issue with running!
./scripts/run_mf.sh 
command, 
I run this on Google collab but I had a strange issue: 
https://github.com/yourh/DeepGraphGO/blob/efdb1cb9425f4f48e4613c0a89e603f5542bcb19/preprocessing.py#L34
Can you please explain the use of ':=' in line 34 in the processing script? 
I added blast as suggested above but it is not working, you can find below a trace of the issue.
Thank you!
![image](https://user-images.githubusercontent.com/71659848/194107221-53bb0609-553e-401c-a044-242c0d4d2789.png)

**Comment by yourh** [comment]
@Amels404 The python version you used is too low. Please use 3.8+ as README said and run preprocessing.py again.

**Comment by Amels404** [comment]
Thanks, I managed to make it work, the issue was in the memory, I only have 32G RAM on my PC. 
I have now another issue related to the Cuda version. I have tried to use DGL with the 11.8 Cuda version (it is not supported) I was wondering which version are you using for Cuda. 
For my side, I installed the version of DGL working with 11.6 but it generates an error. 
![image](https://user-images.githubusercontent.com/71659848/194341010-77caa15f-2623-4d90-9bfd-c3948d001f43.png)
